src/components/application_manager/src/rpc_service_impl.cc

```diff
@@ -392,7 +392,9 @@ bool RPCServiceImpl::ManageHMICommand(const commands::MessageSharedPtr message,
   if (kRequest == message_type) {
     SDL_LOG_DEBUG("ManageHMICommand");
     command->set_warning_info(warning_info);
-    request_ctrl_.AddHMIRequest(command);
+    if (!app_manager_.IsStopping()) {
```

Comment:
Hello, can you please explain this change a bit?

Why would this only cover `AddHMIRequest`? Is this change necessary to prevent a crash? Why would this change not exit the method much earlier like the low voltage check?

Reply:
Hi!
>> Is this change necessary to prevent a crash?
This changes is neccesary here as HMI Commands gets corrupted if they being added to Request controller after ApplicationManger is requested to stop.
>>Why would this only cover AddHMIRequest?
HMI requests are stored in a separate collection in Request Controller and crash is related exactly to this collection.
>>Why would this change not exit the method much earlier like the low voltage check?
SDL need to send requests to HMI during stopping(e.g. Navigation.StopAudioStream, Navigation.StopStream reuqests), but in this case we don't need to wait for responses. By this reason we need to call Run() method, but we don't need to track it with Request controller.